### PR TITLE
fix: improve CM modified date comparison

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?
Remove the clunky date check for Modified status, now , when publishing on the document service, we assign the updated value of the published version to be exactly the same as the publish version.

Before they were not "exactly" the same, (off by milliseconds) and we had to add a threshold for the check

Reopened from https://github.com/strapi/strapi/pull/19787

### CODE REVIEW

**Comment by jhoward1994** [comment]
Thanks! Makes sense to me.

I've merged this into https://github.com/strapi/strapi/pull/20258 to check the CI on the tests I was having trouble with. I'll approve if everything works

**Comment by jhoward1994** [comment]
Okay thanks.

I'm working on it a bit in https://github.com/strapi/strapi/pull/20258. Sorry it breaks our PR organisation a bit but I need to test it with the e2e tests in that branch. I can cherry pick it here so we can trace it all

File: packages/core/content-manager/server/src/services/document-metadata.ts [hunk 1]
```diff
@@ -155,28 +149,46 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
...
+      } else {
+        draftVersion = otherVersion;
+      }
+    }
+
     if (!otherDocumentStatuses?.length) {
-      // It there are no other versions we take the current version status
+      // If there are no other versions, the current version is the latest
       return isDraft ? CONTENT_MANAGER_STATUS.DRAFT : CONTENT_MANAGER_STATUS.PUBLISHED;
     }
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/services/document-metadata.ts:175]: nice, was toying with it, could we simplify a bit the comparison to check if there is a draft version or published version available?
```ts
    if (isDraft) {
      draftVersion = version;
    } else {
      publishedVersion = version;
    }
	
     const otherVersion = otherDocumentStatuses?.at(0);
      if (otherVersion?.publishedAt) {
        publishedVersion = otherVersion;
      } else if (otherVersion) {
        draftVersion = otherVersion;
      }

    if(!draftVersion) return published
    if (!publishedVersion) return draft
    
    const isDraftModified = getIsVersionLatestModification(draftVersion, publishedVersion);
    return isDraftModified ? CONTENT_MANAGER_STATUS.MODIFIED : CONTENT_MANAGER_STATUS.PUBLISHED;
```

**Review by Marc-Roig — COMMENTED** [review]
nice! thank you for the work Jamie, lets push to fix all api tests and we can merge this. Let me know if we want me to jump in a call at any time!

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**